include/xtensor/xlayout.hpp

```diff
@@ -0,0 +1,22 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille and Sylvain Corlay                     *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XLAYOUT_HPP
+#define XLAYOUT_HPP
+
+namespace xt
+{
+    enum class layout
+    {
+        row_major,
+        column_major,
+        dynamic
+    };
+}
+
+#endif
```

Comment:
Actually starting with C++11 you can forward declare an enum as long as you provide the storage type. But since you need to provide a default layout for `xtensor` and `xarray` aliases, you need to know the enum definition in `xtensor_forward.hpp`. So it might be relevant to define it here.